### ISSUE TITLE
Changing testing lead on engagement is not working

### ISSUE THREAD

**Issue by madchap** [DefectDojo/django-DefectDojo#4058]
**Bug description**
Cannot edit and save an engagement with a `Testing lead` that's not `admin` or possibly another staff/superuser user.

Going through the API works.

**Steps to reproduce**
1. Create a powerless user
2. Go to an engagement and edit it
3. Try to assign the testing lead role to powerless user (even if he's authz to the product or type, does not matter)
3. Notice the edit page reloads and your changes are gone. No errors are given.

**Expected behavior**
Edits to the engagement are saved.

**Deployment method** *(select with an `X`)*
- [x] Kubernetes
- [x] Docker
- [ ] setup.bash / legacy-setup.bash

Latest `dev`.

**Comment by valentijnscholten** [comment]
FWIW both cases (if I understand them correctly) work find for me on latest `dev`.

**Comment by madchap** [comment]
It seems that the local `admin` user can always save as well as staff (maybe superadmin).

But a regular "active" user with authz to the product, no error nowhere. The edit screen just reloads. I could confirm on the latest dev branch.

The user also is granted "product types" authorization if this matters.

**Comment by madchap** [comment]
In the middle of some jira debug, I see this:

```
uwsgi_1         | [18/Mar/2021 15:11:23] DEBUG [dojo.engagement.views:224] <ul class="errorlist"><li>lead<ul class="errorlist"><li>Select a valid choice. That choice is not one of the available choices.</li></ul></li></ul>
ment: egw (Mar 17, 2021)
```

**Comment by madchap** [comment]
Adjusted the description.

**Comment by valentijnscholten** [comment]
Sounds like the same problem that it's fixed in #4078

**Comment by valentijnscholten** [comment]
fixed in dev / 1.14.0
